check_powerdns_rec.py

```diff
@@ -188,86 +290,56 @@ def calc_avgps(_data_old, _data_new):
-                 "dont-outqueries\t0\nedns-ping-matches\t0\nedns-ping-mismatches\t0\nfailed-host-entries\t0\n"\
-                 "ipv6-outqueries\t0\nipv6-questions\t0\nmalloc-bytes\t0\nmax-mthread-stack\t0\nnegcache-entries\t0\n"\
-                 "no-packet-error\t0\nnoedns-outqueries\t0\nnoerror-answers\t0\nnoping-outqueries\t0\n"\
-                 "nsset-invalidations\t0\nnsspeeds-entries\t0\nnxdomain-answers\t0\noutgoing-timeouts\t0\n"\
-                 "over-capacity-drops\t0\npacketcache-entries\t0\npacketcache-hits\t0\npacketcache-misses\t0\n"\
-                 "policy-drops\t0\nqa-latency\t0\nquestions\t0\nresource-limits\t0\nserver-parse-errors\t0\n"\
-                 "servfail-answers\t0\nspoof-prevents\t0\nsys-msec\t0\ntcp-client-overflow\t0\ntcp-clients\t0\n"\
-                 "tcp-outqueries\t0\ntcp-questions\t0\nthrottle-entries\t0\nthrottled-out\t0\n"\
-                 "throttled-outqueries\t0\nunauthorized-tcp\t0\nunauthorized-udp\t0\nunexpected-packets\t0\n"\
-                 "unreachables\t0\nuptime\t0\nuser-msec\t0\nsecurity-status\t1\n"
-
-        data_new = parse_pdns(stdout)
-        data_new['epoch'] = int(time.time())
-
-        data_old = data_new.copy()
-        data_old['epoch'] -= 1
-
-        (data_avg, queries) = calc_avgps(data_old, data_new)
-    else:
-        try:
-            cli = [pdns_tool]
-            if args.socket_dir:
-                cli.append('--socket-dir=%s' % args.socket_dir)
-            if args.config_name != '':
-                cli.append('--config-name=%s' % args.config_name)
-            cli.append('get-all')
-
-            MyOut = subprocess.Popen(cli, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-            stdout, stderr = MyOut.communicate()
-            if MyOut.returncode != 0:
-                monitor.set_message(stdout)
-                monitor.report()
-        except OSError:
-            monitor.set_message("Control command '%s' not found." % pdns_tool)
-            monitor.report()
-
-        # noinspection PyUnboundLocalVariable
-        data_new = parse_pdns(stdout)
-        data_new['epoch'] = int(time.time())
-
-        filename = get_fname(args.scratch, args.config_name)
-        data_old = load_measurement(filename)
+    try:
+        args = parse_args()
 
+        if args.test:
+            data_old, data_new = fake_statistics()
+        else:
+            if args.api_host:
+                pdns = PowerDnsApi(args.api_host, args.api_port, args.api_key)
+                result = pdns.statistics()
+                data_new = parse2_pdns(result)
+            else:
+                pdns = PowerDnsCtrlTool(args.socket_dir, args.config_name)
+                result = pdns.get_all()
+                data_new = parse_pdns(result)
+            # noinspection PyUnboundLocalVariable
+            data_new['epoch'] = int(time.time())
+            filename = get_fname(args.scratch, args.config_name)
+            data_old = load_measurement(filename)
```

Comment:
I think `load_measurement` should throw an exception in case the filename isn't there (instead of returning an empty dict). We could try/expect this scenario here (and continue/pass in this first-run case).